### ISSUE TITLE
DLL load failed on Appveyor Windows

### ISSUE THREAD

**Issue by djhoese** [conda-forge/gdal-feedstock#241]
I have a project that is using astropy's [ci-helpers](https://github.com/astropy/ci-helpers) to build a conda environment on appveyor. Nothing has been changed by us in our build setup, but we are now getting a DLL load failed error message:

```
  File "c:\projects\satpy\satpy\writers\geotiff.py", line 30, in <module>
    from osgeo import gdal, osr
  File "C:\Python27_64\envs\test\lib\site-packages\osgeo\__init__.py", line 25, in <module>
    _gdal = swig_import_helper()
  File "C:\Python27_64\envs\test\lib\site-packages\osgeo\__init__.py", line 21, in swig_import_helper
    _mod = imp.load_module('_gdal', fp, pathname, description)
ImportError: DLL load failed: The specified module could not be found.
```

You can see the full job and environment here: https://ci.appveyor.com/project/pytroll/satpy/builds/20702760/job/x0mo4l3940ff4n9h?fullLog=true

As far as I can tell both libgdal, gdal, and kealib are all coming from conda-forge. Any other ideas to what has changed recently that would cause this?

**Comment by djhoese** [comment]
Yes, it is done automatically by `ci-helpers`. You can see the path to the environment packages in the error message: `C:\Python27_64\envs\test`

**Comment by mingwandroid** [comment]
Given we see: https://github.com/astropy/ci-helpers/blob/6ac17697e48002ec74bd8ef98bf6376a475a5059/appveyor/install-miniconda.ps1#L288

I suspect activation doesn't work right here. I don't know anything about powershell though so can't be of much more help.

**Comment by gillins** [comment]
BTW we no longer support Python 2.7 on Windows (`gdal` won't compile on MSVC 2008)... Maybe there should be a better error message about this?

**Comment by djhoese** [comment]
Thanks @gillins. As long as there are older versions available on conda-forge until the end of 2019 I shouldn't be too affected.

**Comment by djhoese** [comment]
I went back to https://github.com/astropy/ci-helpers/commit/4de395704381fff34c3a104bb35e1a5e0f654d9e#diff-567199671add953f07327d2790dbb56a

And still have the error. I compared the conda package versions from the last successful run from a day ago and the biggest difference I noticed is krb5 going from 1.14.6 to 1.16.2. Trying to force the version now and see if that changes anything.

**Comment by djhoese** [comment]
Progress! I copied `.../my_conda_env/Library/bin/tiff.dll` to `.../my_conda_env/Library/bin/libtiff.dll` and things seem to be importable on my VM. Will try this hack on the appveyor environment.

**Comment by ocefpaf** [comment]
> @ocefpaf I know you've helped me with linux linking issues with gdal/rasterio in the past. Do you have any ideas on this one?

Sorry but Windows is not my thing. @gillins probably knows better.

PS: this issue is probably related to #242

**Comment by djhoese** [comment]
@gillins @ocefpaf I've figured out the issue for my environments. For the last couple builds of `libtiff` the shared library on Windows is called `tiff.dll`, but gdal seems to be built with one of the older builds which uses `libtiff.dll`. If I add a small hack to my `appveyor.yaml` that copies `tiff.dll` to `libtiff.dll` then gdal imports as expected:

```bash
python -c "from osgeo import gdal, osr"
```

I'll file a bug with the libtiff feedstock with further details unless you can quickly reply to this letting me know that this is an expected change for libtiff.

**Comment by ocefpaf** [comment]
> I'll file a bug with the libtiff feedstock with further details unless you can quickly reply to this letting me know that this is an expected change for libtiff.

I can reproduce it in #243 but the odd thing is that `libtiff` is not changed in ages :-/ (I'll take a closer look into that tomorrow.)

**Comment by djhoese** [comment]
Thanks for the info @ocefpaf. See https://github.com/conda-forge/libtiff-feedstock/issues/35 for some things I've been looking at and some wild guesses I have.

**Comment by kmuehlbauer** [comment]
@djhoese @ocefpaf AFAIK, gdal has an internal copy of libtiff. Could it be, that the gdal build process has changed somehow, using internal libtiff copy. There have been recent changes to libtiff I read on the gdal mailing list.

**Comment by djhoese** [comment]
@kmuehlbauer The conda-forge recipe at least has depended on `libtiff` and `geotiff` for some time: https://github.com/conda-forge/gdal-feedstock/blob/master/recipe/meta.yaml#L47

Also note that older versions of gdal (not just the new builds/versions) like v2.2.4 also fail with the most recent builds of libtiff (non-vc9/vc14).

**Comment by djhoese** [comment]
The question is, is the libtiff.dll -> tiff.dll desired by libtiff? Is that a bug or unexpected result from a change in conda-build? If gdal was built against the new builds of libtiff (tiff.dll), would it have picked up `tiff.dll`? Is that a limitation of gdal's build scripts that it didn't use `tiff.dll` but should be able to? I guess it depends on what is considered standard practice for libraries on Windows and if the libtiff name change was desired.

**Comment by mingwandroid** [comment]
> Is that a bug or unexpected result from a change in conda-build

Never say never of course but i think it very unlikely that anything in conda-build would have any bearing on this issue.

**Comment by ocefpaf** [comment]
> Never say never of course but I think it very unlikely that anything in conda-build would have any bearing on this issue.

BTW, @djhoese already found the issue and it is from when I moved the recipe to be more like AR and removed a workaround we had in place for it. B/c of that believe the right "fix" is in `gdal` and not `libtiff`.
